### PR TITLE
Use sideload based snapshots in arb-rpc-node

### PR DESCRIPTION
**Pull request by PlasmaPower**
Currently breaks tests as the arb-rpc-node tests don't use arbcore, which is required for the sideload based approach

### CODE REVIEW

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp [hunk 1]
```diff
@@ -436,9 +436,15 @@ rocksdb::Status ArbCore::reorgToMessageOrBefore(
         pending_checkpoint = Checkpoint{};
     }
 
+    auto status =
```
Comment (hkalodner) [inline packages/arb-avm-cpp/data_storage/src/arbcore.cpp:439]: Seems like there might be an off by one issue here. This checkpoint is still valid so it should be `pending_checkpoint.block_height` I think

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp [hunk 2]
```diff
@@ -1936,8 +1981,11 @@ ValueResult<std::unique_ptr<Machine>> ArbCore::getMachineForSideload(
     // Check the cache
     {
         std::shared_lock<std::shared_mutex> lock(sideload_cache_mutex);
-        auto it = sideload_cache.find(block_number);
-        if (it != sideload_cache.end()) {
+        // Look for the first value after the value we want
```
Comment (hkalodner) [inline packages/arb-avm-cpp/data_storage/src/arbcore.cpp:1984]: I think this should be the first value `<= block_number`

**Review by hkalodner — CHANGES_REQUESTED** [review]

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp (hunk 2 above)
Comment (PlasmaPower) [reply packages/arb-avm-cpp/data_storage/src/arbcore.cpp:1984]: It should be, but C++'s APIs are stupid here and don't provide such a method, so instead we go to the first value greater than the block number, and then go back one.

Comment (hkalodner) [reply packages/arb-avm-cpp/data_storage/src/arbcore.cpp:1984]: Ahh, I read the code to fast. Makes sense

**Review by hkalodner — APPROVED** [review]
This looks good to me pending getting the arbostest tests working

**Pull request merged**